Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -7607,7 +7607,7 @@ void HTMLMediaElement::didStopBeingFullscreenElement()
     m_changingVideoFullscreenMode = false;
 }
 
-#if ENABLE(FULLSCREEN_API)
+#if ENABLE(FULLSCREEN_API) && ENABLE(FULLSCREEN_VIDEO_PREVENT_HDR_SUPPRESSION)
```

Comment:
This doesn't seem right, but maybe it is. Why should only platforms with `ENABLE(FULLSCREEN_VIDEO_PREVENT_HDR_SUPPRESSION)` have `documentFullscreenChanged`? At least `m_isChildOfElementFullscreen` doesn't seem very HDR-specific.

Reply:
It's because passing along the full screen state is only needed in situations where it _may_ influence the suppress-HDR handling. So on non-Mac platforms, it would take unnecessary work and space to pass and store this value, as it would never be read.
I was hoping that having both `FULLSCREEN` and `HDR` in the macro name would convey that special connection here.
Now, this event should be fairly rare (only when full screen state changes), and the storage is only 1 byte, so it may be okay not to optimize this. Other opinions welcome.

Reply:
That makes sense, just wanted to double check. Looks good to me with the other comments addressed.